docs/whitepapers/kalmanfilter/kalmanfilter.tex

```diff
@@ -0,0 +1,345 @@
+% \[ K_t = P_{t} C^T (C_t P_t C^T + R)^{-1} \]
+% where \( R_t \) is the measurement noise covariance matrix.
+
+% \subsection{State Update}
+% Update the state estimate with the new measurement by applying the Kalman Gain to the innovation:
+% \[ \widetilde{x}_{t+1} = \widetilde{x}_{t} + K_t ( y_t + - C\widetilde{x}_t)\]
+
+% \subsection{Covariance Update}
+% Update the estimate's error covariance to reflect the decrease in uncertainty:
+% \[ P_{t+1} = (I - K_t C) P_{t} \]
+% where \( I \) is the identity matrix.
+
+
+\newpage
+\section{altitude dynamics}
+The altitude dynamics of a quadcopter are defined within a global coordinate system, crucial for maintaining a predetermined altitude from the Earth's surface. The model that describes these dynamics is based on fundamental principles, delineated as follows:
+\\
+
+\noindent
+The rate of change of the quadcopter's altitude, represented as \( \dot{z}_t \), is the result of the vertical acceleration \( a^z_{T_t} \) produced by the drone's motors at a given time minus the gravitational acceleration, \( g \). This equation is continuous in time and is expressed as:
+
+\begin{equation}
+\dot{z}_t = a^z_{T_t} - g
+\end{equation}
+
+\noindent
+Here, \( a^z_{T_t} \) signifies the upward acceleration generated by the propulsion at time \( t \), measured in meters per second squared. The constant \( g \) denotes the acceleration due to Earth's gravity, also in meters per second squared. The altitude \( z_t \) represents the drone's center of mass's vertical position at time \( t \), measured in meters.
+\\
+
+\noindent
+Additionally, the drone's vertical velocity \( v_{z_t} \) and vertical acceleration \( a_{z_t} \) are defined by the rate of altitude change \( \dot{z}_t \) and the rate of vertical acceleration change \( \dot{a}^z_{T_t} - g \), respectively. The term \( \dot{a}^z_{T_t} \) is derived from the quadcopter's upward thrust and serves as the system's input, while \( g \) is considered a constant input in the opposite direction.
+\\
+
+\noindent
+Let \( y_{t}^z = z_t \) be the output equation of the system. Then the corresponding state-space representation is,
+
+\begin{equation}
+\begin{bmatrix}
+v_{v,t}\\
+a_{t}^z
+\end{bmatrix} =
+\begin{bmatrix}
+0 & 1 \\
+0 & 0 
+\end{bmatrix}
+\begin{bmatrix}
+z\\
+v_{v,t}
+\end{bmatrix} + 
+\begin{bmatrix}
+0 & 0 \\
+1 & 1 \\ 
+\end{bmatrix}
+\begin{bmatrix}
+a_{T,t}^z \\ 
+-g
+\end{bmatrix},
+\end{equation}
+\\
+
+\noindent
+for the Barometer sensor's output, denoted by \( y_{barom} \), is described by the equation:
+\begin{equation}
+y^{barom} = z + bias + V^{barom}
+\end{equation}
+\\
+where, 
+\begin{equation}
+bias_{t+1} = bias_t + W_{t}^{bias}
+\end{equation}
+the bias should stay consistent throughout readings 
+\\
+\\ 
+
+
+\noindent
+for the GPS sensor's output, denoted by \( y_{gps} \), is described by the equation:
```

Comment:
Decide whether `gps`, `barometer`, etc will be superscripts or subscripts. Use `{\rm gps}` in either case.